tests/unit/test_tf_dataloader.py

```diff
@@ -272,3 +274,93 @@ def test_validater(tmpdir, batch_size):
     true_auc = roc_auc_score(labels, predictions)
     estimated_auc = logs[auc_key]
     assert np.isclose(true_auc, estimated_auc, rtol=1e-6)
+
+
+@pytest.mark.parametrize("engine", ["parquet"])
+@pytest.mark.parametrize("batch_size", [1, 10, 100])
+@pytest.mark.parametrize("global_rank", [0, 1])
+def test_multigpu_partitioning(datasets, engine, batch_size, global_rank):
+    cont_names = ["x", "y", "id"]
+    cat_names = ["name-string", "name-cat"]
+    label_name = ["label"]
+    data_loader = tf_dataloader.KerasSequenceLoader(
+        str(datasets["parquet"]),
+        cat_names=cat_names,
+        cont_names=cont_names,
+        batch_size=batch_size,
+        buffer_size=0.1,
+        label_names=label_name,
+        engine=engine,
+        shuffle=False,
+        global_size=2,
+        global_rank=global_rank,
+    )
+    indices = data_loader._gather_indices_for_dev(None)
+    assert indices == [global_rank]
+
+
+hvd = pytest.importorskip("horovod")
+
+
+def test_hvd(tmpdir):
+
+    json_sample = {
+        "conts": {},
+        "cats": {
+            "genres": {
+                "dtype": None,
+                "cardinality": 50,
+                "min_entry_size": 1,
+                "max_entry_size": 5,
+                "multi_min": 2,
+                "multi_max": 4,
+                "multi_avg": 3,
+            },
+            "movieId": {
+                "dtype": None,
+                "cardinality": 500,
+                "min_entry_size": 1,
+                "max_entry_size": 5,
+            },
+            "userId": {"dtype": None, "cardinality": 500, "min_entry_size": 1, "max_entry_size": 5},
+        },
+        "labels": {"rating": {"dtype": None, "cardinality": 2}},
+    }
+    cols = datagen._get_cols_from_schema(json_sample)
+    df_gen = datagen.DatasetGen(datagen.UniformDistro(), gpu_frac=0.0001)
+    target_path = os.path.join(tmpdir, "input/")
+    os.mkdir(target_path)
+    df_files = df_gen.full_df_create(10000, cols, output=target_path)
+    # process them
+    cat_features = nvt.ColumnGroup(["userId", "movieId", "genres"]) >> nvt.ops.Categorify()
+    ratings = nvt.ColumnGroup(["rating"]) >> (lambda col: (col > 3).astype("int8"))
+    output = cat_features + ratings
+    proc = nvt.Workflow(output)
+    train_iter = nvt.Dataset(df_files, part_size="10MB")
+    proc.fit(train_iter)
+    target_path_train = os.path.join(tmpdir, "train/")
+    os.mkdir(target_path_train)
+    proc.transform(train_iter).to_parquet(output_path=target_path_train, out_files_per_proc=5)
+    # add new location
+    target_path = os.path.join(target_path_train, "workflow/")
+    os.mkdir(target_path)
+    proc.save(target_path)
+    process = subprocess.Popen(
+        [
+            "horovodrun",
+            "-np",
+            "2",
+            "-H",
+            "localhost:2",
+            "./hvd_wrapper.sh",
```

Comment:
We should probably use absolute paths - rather than paths relative to the repo root here.